pkg/controller/tenants.go

```diff
@@ -45,6 +60,162 @@ func (c *Controller) getTenantConfiguration(ctx context.Context, tenant *miniov2
 	return tenantConfiguration, nil
 }
 
+func (c *Controller) saveTenantConfiguration(ctx context.Context, tenant *miniov2.Tenant) error {
+	prevConfiguration, err := c.getTenantConfiguration(ctx, tenant)
+	if err != nil {
+		return fmt.Errorf("secret '%s' reference on tenant.spec.configuration not found: %v", tenant.Spec.Configuration, err)
+	}
+	tenantConfiguration := map[string]string{}
+	if _, ok := prevConfiguration[storageClassEnv]; ok {
+		tenantConfiguration[storageClassEnv] = string(prevConfiguration[storageClassEnv])
+	}
+
+	if _, ok := prevConfiguration[rootUserEnv]; ok {
+		tenantConfiguration[rootUserEnv] = string(prevConfiguration[rootUserEnv])
+	}
+
+	if _, ok := prevConfiguration[rootUserPassword]; ok {
+		tenantConfiguration[rootUserPassword] = string(prevConfiguration[rootUserPassword])
+	}
+
+	if _, ok := prevConfiguration[accessKeyEnv]; ok {
+		tenantConfiguration[accessKeyEnv] = string(prevConfiguration[accessKeyEnv])
+	}
+
+	if _, ok := prevConfiguration[secretKeyEnv]; ok {
+		tenantConfiguration[secretKeyEnv] = string(prevConfiguration[secretKeyEnv])
+	}
+
+	if _, ok := prevConfiguration[minioBrowserEnv]; ok {
+		tenantConfiguration[minioBrowserEnv] = string(prevConfiguration[minioBrowserEnv])
+	}
+
+	// Update credentials  based on the existing credsSecret, we will overwrite root credentials only when the fields
+	// accesskey and secretkey are not empty
+	if tenant.HasCredsSecret() {
+		credsSecret, err := c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Get(ctx, tenant.Spec.CredsSecret.Name, metav1.GetOptions{})
+		if err != nil && !k8serrors.IsNotFound(err) {
+			return err
+		}
+		var accessKey string
+		var secretKey string
+		if _, ok := credsSecret.Data["accesskey"]; ok {
+			accessKey = string(credsSecret.Data["accesskey"])
+		}
+		if _, ok := credsSecret.Data["secretkey"]; ok {
+			secretKey = string(credsSecret.Data["secretkey"])
+		}
+		if accessKey != "" || secretKey != "" {
+			tenantConfiguration["MINIO_ROOT_USER"] = accessKey
+			tenantConfiguration["MINIO_ROOT_PASSWORD"] = secretKey
+		}
+	}
+
+	// Enable `mc admin update` style updates to MinIO binaries
+	// within the container, only operator is supposed to perform
+	// these operations.
+	tenantConfiguration["MINIO_UPDATE"] = "on"
+	tenantConfiguration["MINIO_UPDATE_MINISIGN_PUBKEY"] = "RWTx5Zr1tiHQLwG9keckT0c45M3AGeHD6IvimQHpyRywVWGbP1aVSGav"
+	tenantConfiguration["MINIO_OPERATOR_VERSION"] = c.operatorVersion
+	tenantConfiguration["MINIO_PROMETHEUS_JOB_ID"] = tenant.PrometheusConfigJobName()
+
+	var domains []string
+	// Enable Bucket DNS only if asked for by default turned off
```

Comment:
`%s/asked for by default/asked for else by default/g`

Reply:
just rephrased it